modules/metrics/src/main/java/org/apache/ignite/internal/metrics/MetricProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.metrics;
+
+import java.util.Map;
+
+/**
+ * Read-only metrics registry.
+ */
+public class MetricProvider {
```

Comment:
Actually, I don't understand the purpose of this entity. It just delegate calls to the `MetricRegistry`. It's also could be usefull to have access to the metrics sources because the class of metrics source could be used as metrics set schema identifier.

Reply:
This wrapper is needed only for `MetricExporter` usage to prodive the clear read-only interface for all available metrics.

I think, that exporters shouldn't have direct access to MetricRegistry itself, it will produce the mess in design and give exporters dangerous possibilities to register new sources and etc.

Reply:
Okay. But exporters will need to have access to RO collection of metric sources.